lib/protobuf/nats/client.rb

```diff
@@ -100,9 +118,12 @@ def nats_request_with_two_responses(subject, data, opts)
 
           # Wait for reply
           first_message = nats.next_message(sub, ack_timeout)
-          fail ::NATS::IO::Timeout if first_message.nil?
+          return :ack_timeout if first_message.nil?
+          return :nack if first_message.data == ::Protobuf::Nats::Messages::NACK
+          return first_message.data unless first_message.data == ::Protobuf::Nats::Messages::ACK
```

Comment:
This should break a test. We need to have an ACK and a response message in any order and we check for that below.

EDIT: I see you changed behavior in the tests... don't think this is a good idea.